### ISSUE TITLE
[Bug]: Client sent an HTTP request to an HTTPS server.

### ISSUE THREAD

**Issue by gaoyuan5251** [milvus-io/milvus#28032]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster): standalone 2.3.1
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): 
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

With Milvus-operator, deploy milvus single-node mode, but rely on Etcd and minio external clusters.

### Expected Behavior

milvus standalone should be running.

### Steps To Reproduce

```markdown
1. vim milvs-external.yaml
2. kubectl apply -f milvs-external.yaml
3. kubectl describe milvus >desc.txt
4.kubectl logs hll-release-milvus-standalone-d57df84d8-qdmms > log.txt

### Milvus Log

[milvs-external.txt](https://github.com/milvus-io/milvus/files/13213385/milvs-external.txt)
[desc.txt](https://github.com/milvus-io/milvus/files/13213398/desc.txt)
[log.txt](https://github.com/milvus-io/milvus/files/13213393/log.txt)



### Anything else?

**Comment by yanliang567** [comment]
looks like the config yaml is not parsed correctly. @haorenfsa could you please take a look?
/assign @haorenfsa 
/unassign

**Comment by gaoyuan5251** [comment]
@haorenfsa @yanliang567 bro, is anyone looking at this issue?

**Comment by haorenfsa** [comment]
@gaoyuan5251 Sry for the delay, cloud you provide the output of `kubectl get configmap hll-release -o yaml`?

**Comment by gaoyuan5251** [comment]
@haorenfsa ok,
[configmap.txt](https://github.com/milvus-io/milvus/files/13262243/configmap.txt)

**Comment by gaoyuan5251** [comment]
@haorenfsa cc, kubectl describe  milvus  hll-release > status.txt
[status.txt](https://github.com/milvus-io/milvus/files/13262257/status.txt)

**Comment by haorenfsa** [comment]
> Is this because the test machine is out of memory?

looks no.

> @haorenfsa cc, kubectl describe milvus hll-release > status.txt [status.txt](https://github.com/milvus-io/milvus/files/13262257/status.txt)

@yanliang567 looks the configmap is rendered correctly. @zhagnlu have you met this error before?

**Comment by gaoyuan5251** [comment]
> > Is this because the test machine is out of memory?
> 
> looks no.
> 
> > @haorenfsa cc, kubectl describe milvus hll-release > status.txt [status.txt](https://github.com/milvus-io/milvus/files/13262257/status.txt)
> 
> @yanliang567 looks the configmap is rendered correctly. @zhagnlu have you met this error before?

...
Status:
  Components Deploy Status:
    Standalone:
      Generation:  1
      Image:       milvusdb/milvus:v2.3.1
      Status:
        Conditions:
          Last Transition Time:  2023-11-06T03:23:34Z
          Last Update Time:      2023-11-06T03:23:34Z
          **Message:               Deployment does not have minimum availability.**
          Reason:                MinimumReplicasUnavailable
          Status:                False
          Type:                  Available
          Last Transition Time:  2023-11-06T03:33:35Z
          Last Update Time:      2023-11-06T03:33:35Z
          Message:               ReplicaSet "hll-release-milvus-standalone-64b987fcf5" has timed out progressing.
          Reason:                ProgressDeadlineExceeded
          Status:                False
          Type:                  Progressing
        Observed Generation:     1
        Replicas:                1
        Unavailable Replicas:    1
        Updated Replicas:        1
        ...
What does this part of the message mean? @haorenfsa

**Comment by zhagnlu** [comment]
https://github.com/milvus-io/milvus/issues/27853  same error. check deploy and config params about k8s

**Comment by gaoyuan5251** [comment]
> #27853 same error. check deploy and config params about k8s
bro, Could you mind help me check, what is wrong with my yaml file?
[milvs-external.yml.txt](https://github.com/milvus-io/milvus/files/13276272/milvs-external.yml.txt)

**Comment by gaoyuan5251** [comment]
> > #27853 same error. check deploy and config params about k8s
> > bro, Could you mind help me check, what is wrong with my yaml file? @zhagnlu 
> > [milvs-external.yml.txt](https://github.com/milvus-io/milvus/files/13276272/milvs-external.yml.txt)

**Comment by gaoyuan5251** [comment]
kubectl logs hll-release-milvus-standalone-6c747c8948-kqdjl > logs.txt
[logs.txt](https://github.com/milvus-io/milvus/files/13276511/logs.txt)

[2023/11/07 06:33:46.053 +00:00] [ERROR] [querynode/service.go:129] ["QueryNode init error: "] [error="Endpoint url cannot have fully qualified paths."] [stack="github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:129\ngithub.com/milvus-io/milvus/internal/distributed/querynode.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:223\ngithub.com/milvus-io/milvus/cmd/components.(*QueryNode).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/query_node.go:53\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:113"]
[2023/11/07 06:33:46.053 +00:00] [ERROR] [components/query_node.go:54] ["QueryNode starts error"] [error="Endpoint url cannot have fully qualified paths."] [stack="github.com/milvus-io/milvus/cmd/components.(*QueryNode).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/query_node.go:54\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:113"]
panic: Endpoint url cannot have fully qualified paths.

@zhagnlu idk why this error is reported

**Comment by xiaofan-luan** [comment]
@gaoyuan5251 
Did you enable SSL on your minio or S3?

**Comment by gaoyuan5251** [comment]
> @gaoyuan5251 Did you enable SSL on your minio or S3?

Nope @xiaofan-luan 
<img width="604" alt="image" src="https://github.com/milvus-io/milvus/assets/38758293/d0fb6b67-b548-4ecb-a426-eb3896b4a74b">

**Comment by xiaofan-luan** [comment]
@zhagnlu @zwd1208

**Comment by xiaofan-luan** [comment]
> > @gaoyuan5251 Did you enable SSL on your minio or S3?
> 
> Nope @xiaofan-luan <img alt="image" width="604" src="https://user-images.githubusercontent.com/38758293/280946919-d0fb6b67-b548-4ecb-a426-eb3896b4a74b.png">

<img width="540" alt="image" src="https://github.com/milvus-io/milvus/assets/83447078/2d68c0ee-d007-43c6-9d33-491568c6b714">
you are not enable ssl, what about your minio config?

**Comment by xiaofan-luan** [comment]
https://github.com/zhagnlu/milvus-store-tool  

this tool could help you to verify your S3 connectivity

**Comment by gaoyuan5251** [comment]
@xiaofan-luan  I uploaded my yaml file at the top so you can take a look

**Comment by haorenfsa** [comment]
@gaoyuan5251 Cloud you use the minio client command [mc](https://min.io/docs/minio/linux/reference/minio-mc.html) to check the if the minio server address `10.102.235.153:80` is available?

**Comment by gaoyuan5251** [comment]
> @gaoyuan5251 Cloud you use the minio client command [mc](https://min.io/docs/minio/linux/reference/minio-mc.html) to check the if the minio server address `10.102.235.153:80` is available?

mc tree hllminio
hllminio
└─ hll-milvus
   └─ k8s部署
   
   @haorenfsa

**Comment by gaoyuan5251** [comment]
> @gaoyuan5251 Cloud you use the minio client command [mc](https://min.io/docs/minio/linux/reference/minio-mc.html) to check the if the minio server address `10.102.235.153:80` is available?

我重新部署了minio集群，cluster-ip变更了。但是可以看到 minio 服务使用上是没问题的。

mc config host list
gcs     
  URL       : https://storage.googleapis.com
  AccessKey : YOUR-ACCESS-KEY-HERE
  SecretKey : YOUR-SECRET-KEY-HERE
  API       : S3v2
  Path      : dns

hllminio
  URL       : http://10.108.111.59:80
  AccessKey : XS1emrJC6zxv65tV
  SecretKey : XS1emrJC6zxv65tV
  API       : s3v4
  Path      : auto

**Comment by haorenfsa** [comment]
@gaoyuan5251 cloud you also get the output of `kubectl exec -it deploy/hll-release-milvus-standalone -- cat /milvus/configs/milvus.yaml`  when the pod is running?

**Comment by gaoyuan5251** [comment]
<img width="1214" alt="image" src="https://github.com/milvus-io/milvus/assets/38758293/c2b1bde4-076e-4192-b5b2-821a0f036974">

**Comment by gaoyuan5251** [comment]
> @gaoyuan5251 cloud you also get the output of `kubectl exec -it deploy/hll-release-milvus-standalone -- cat /milvus/configs/milvus.yaml` when the pod is running?

@haorenfsa 
kubectl exec -it deploy/hll-release-milvus-standalone -- cat /milvus/configs/milvus.yaml
Defaulted container "standalone" out of: standalone, config (init)
error: unable to upgrade connection: container not found ("standalone")

**Comment by haorenfsa** [comment]
you can first delete the old pod `kubectl delete pod  hll-release-milvus-standalone-6c747c8948-kqdjl`, kubernetes will  automatically start a new pod for you. then try this again

`kubectl exec -it deploy/hll-release-milvus-standalone -- cat /milvus/configs/milvus.yaml`

**Comment by gaoyuan5251** [comment]
> you can first delete the old pod `kubectl delete pod hll-release-milvus-standalone-6c747c8948-kqdjl`, kubernetes will automatically start a new pod for you. then try this again

@haorenfsa  done
<img width="1110" alt="image" src="https://github.com/milvus-io/milvus/assets/38758293/d382b993-3f9a-4a24-ac14-bf2c4108ac2b">

**Comment by haorenfsa** [comment]
keep trying, until you got a yaml output ...

**Comment by gaoyuan5251** [comment]
[milvus.yaml.txt](https://github.com/milvus-io/milvus/files/13277287/milvus.yaml.txt)

@haorenfsa done

**Comment by gaoyuan5251** [comment]
@haorenfsa   Isn't the reason obvious? etcd and minio, which build their own clusters, seem to have no problem

**Comment by haorenfsa** [comment]
> @haorenfsa Isn't the reason obvious? etcd and minio, which build their own clusters, seem to have no problem

very strange, I run the same code with your uploaded configuration, and no error occurs

**Comment by haorenfsa** [comment]
@gaoyuan5251 looks like the  k8s `service` `minio` in the same namespace will sets env of `MINIO_PORT` as an unexpected value. try deploy the minio in different namespace with milvus instead.

ref: 
https://github.com/milvus-io/milvus/issues/28215#issuecomment-1798001888

**Comment by haorenfsa** [comment]
Or you can just delete the  k8s `service` `minio` and rename it to other name than just minio, like `minio-service` or so

**Comment by gaoyuan5251** [comment]
> Or you can just delete the k8s `service` `minio` and rename it to other name than just minio, like `minio-service` or so

Sry for the delay, thank u @haorenfsa 
After I move minio to a different namespace in k8s, milvus is worked.
but idk why it work ,hh

**Comment by haorenfsa** [comment]
@gaoyuan5251 it's an issue of the minio SDK used by milvus. The SDK uses the env var `MINIO_PORT` to overwrite its config. You created a service named `minio`, and kubernetes auto generates a `$servicename_PORT`(which is also `MINIO_PORT`)  env, value as `tcp://svcIP:svcPort` for every pod in the same namespace. so your configuration of minio port which is `80` got overwritten by `tcp://svcIP:svcPort`.

So long store short, don't name your MinIO service as `minio`, where ever you uses MinIIO.

**Comment by gaoyuan5251** [comment]
> @gaoyuan5251 it's an issue of the minio SDK used by milvus. The SDK uses the env var `MINIO_PORT` to overwrite its config. You created a service named `minio`, and kubernetes auto generates a `$servicename_PORT`(which is also `MINIO_PORT`) env, value as `tcp://svcIP:svcPort` for every pod in the same namespace. so your configuration of minio port which is `80` got overwritten by `tcp://svcIP:svcPort`.
> 
> So long store short, don't name your MinIO service as `minio`, where ever you uses MinIO.

ok, i get it, thank u so much ~

**Comment by wangjn163** [comment]
> Or you can just delete the k8s `service` `minio` and rename it to other name than just minio, like `minio-service` or so

我用这个解决了问题

**Comment by Niceies** [comment]
@haorenfsa @yanliang567 
# Request to open
This problem is not a problem with minio sdk. minio sdk does not get MINIO_PORT from the environment variable. It is that envsoure is configured to read the environment variable when Milvus is started. However, the KeyFormatter logic that matches the MILVUS environment variable seems to lack a consideration that the environment variable that does not start with MILVUS_ can also be hit.

# pkg/util/paramtable/base_table.go
```
func (bt *BaseTable) init() {
	formatter := func(key string) string {
		ret := strings.ToLower(key)
		ret = strings.TrimPrefix(ret, "milvus.")
		ret = strings.ReplaceAll(ret, "/", "")
		ret = strings.ReplaceAll(ret, "_", "")
		ret = strings.ReplaceAll(ret, ".", "")
		return ret
	}
....
}
```
I think it is correct to filter out the environment variables of milvus when setting the environment variables. After I added the following code, my milvus started normally.
```
func NewEnvSource(KeyFormatter func(string) string) EnvSource {
	es := EnvSource{
		configs:      typeutil.NewConcurrentMap[string, string](),
		KeyFormatter: KeyFormatter,
	}

	for _, value := range os.Environ() {
		rs := []rune(value)
		in := strings.Index(value, "=")
		key := string(rs[0:in])
		value := string(rs[in+1:])
		if !strings.HasPrefix(key, "MILVUS_") {              # Add a variable to determine whether to skip non-milvus
			continue
		}
		envKey := KeyFormatter(key)
		es.configs.Insert(key, value)
		es.configs.Insert(envKey, value)
	}
	return es
}
```

![Image](https://github.com/user-attachments/assets/bbed7e44-48a3-4c80-a34f-8a9cef9008fe)

**Comment by Niceies** [comment]
I am running in k8s mode, and then connecting to the dependent storage components. What I mean is that it is a better way to obtain parameter settings if we only use environment variables starting with the special identifier Milvus to match, otherwise there may be conflicts with system variables.

![Image](https://github.com/user-attachments/assets/f10644fa-2d8c-47d0-818f-bbdf0c5302b1)

At first, I set the externalS3 address through --set, as shown in the figure, but it failed to start, as shown in the title of this issue. Then I found that the problem was that the port was concatenated in the minio configuration initialization address, resulting in an illegal format. In fact, the mino.port here uses the value in the environment variable (tcp://10.192.122.189:9000)

![Image](https://github.com/user-attachments/assets/d51e5976-8283-4002-bd21-94650836f86c)
![Image](https://github.com/user-attachments/assets/f8581de8-779e-4ca4-9167-74ba3f8a6c12)

**Comment by Niceies** [comment]
@xiaofan-luan 
In addition, the minio service appears because the current version of the minio-operator built-in code fixes this content. If you want to deploy minio and milvus in the same namespace, there will inevitably be problems. I tried to start milvus by reading only the configuration file instead of reading the environment variable, but I checked the config.skipEnv option in the milvus code and could not be configured externally.

This is minio-operator code:

![Image](https://github.com/user-attachments/assets/119a5731-c349-420a-a02d-877a26e93921)
